### PR TITLE
[stable] Fix some 2.096 regressions

### PR DESCRIPTION
**Pull request by kinke**

### CODE REVIEW

File: src/dmd/mtype.h [hunk 1]
```diff
@@ -180,6 +182,7 @@ class Type : public ASTNode
     static Type *tdstring;              // immutable(dchar)[]
     static Type *terror;                // for error recovery
     static Type *tnull;                 // for null type
+    static Type *tnoreturn;             // for bottom type typeof(*null)
```
Comment (RazvanN7) [inline src/dmd/mtype.h:185]: Why is this added to `stable`. The implementation of bottom types is currently incomplete in `master` and non-existent in `stable`. Shouldn't we wait until bottom types are fully implemented in master?

**Comment by RazvanN7** [comment]
Are these regressions filed anywhere?

File: src/dmd/mtype.h (hunk 1 above)
Comment (kinke) [reply src/dmd/mtype.h:185]: I've unexpectedly encountered it in the codegen AST, so it's definitely in stable. IIRC, when compiling `core.internal.abort`.

**Comment by kinke** [comment]
I don't file such trivial regressions which are meant to be fixed for the final release and don't seem to affect the 1st DMD beta either.

File: src/dmd/statementsem.d [hunk 2]
```diff
@@ -3712,17 +3712,20 @@ else
             v.dsymbolSemantic(sc);
             cs.push(new ExpStatement(ss.loc, v));
 
-            auto args = new Parameters();
-            args.push(new Parameter(0, t.pointerTo(), null, null, null));
+            auto enterArgs = new Parameters();
+            enterArgs.push(new Parameter(0, t.pointerTo(), null, null, null));
```
Comment (RazvanN7) [inline src/dmd/statementsem.d:3716]: This seems to be just a refactoring. Is this really necessary to backport to stable?

Comment (kinke) [reply src/dmd/statementsem.d:3716]: Please see the commit msg, it's a fix.

Comment (RazvanN7) [reply src/dmd/statementsem.d:3716]: I am not talking about the entire commit, but for this specific renaming from `args` to `enterArgs`. Maybe I am missing something obvious, but to me it just looks like a name change. If we leave the lines 3715 and 3716 untouched, will that invalidate the fix?

Comment (kinke) [reply src/dmd/statementsem.d:3716]: It *is* a name change, emphasizing that it's the param types for the enter-function only, not the exit function too as previously. Sorry but I don't think debating over this is worth our time. ;)

File: src/dmd/mtype.h (hunk 1 above)
Comment (Geod24) [reply src/dmd/mtype.h:185]: Because we keep on getting piece-meal PRs such as https://github.com/dlang/druntime/pull/3381

Comment (RazvanN7) [reply src/dmd/mtype.h:185]: But all those PRs were targeting master, how did bits end up in stable?

Comment (ibuclaw) [reply src/dmd/mtype.h:185]: Because it's 2.096.0beta.

Comment (RazvanN7) [reply src/dmd/mtype.h:185]: Got it.

**Review by RazvanN7 — APPROVED** [review]
I'm approving it with a request that in the future, since the two commits are
completely independent of each other, that they be placed into different PRs.
Thanks!
